### PR TITLE
Emit an event after index status update

### LINKED ISSUE #1942: Emit an event after index status update
### Description
Emit `update.index.status` event after each index status update.

### Motivation
Emit an event after index status update so the user is aware about service status change

### Acceptance Criteria

- Event is emitted every time the status is updated
- All the documentation (README/Lisk docs) are up-to-date
- Tests (if applicable) are written
- Entry is added to the `framework/bin/socket_io_subscribe_client.js` script

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1942 

### How was it solved?
- [x] Event is emitted every time the status is updated
- [x] All the documentation (README/Lisk docs) are up-to-date
- [x] Tests (if applicable) are written
- [x] Entry is added to the framework/bin/socket_io_subscribe_client.js script

### How was it tested?
- [x] Local by running [script](https://github.com/LiskHQ/lisk-service/blob/development/framework/bin/socket_io_subscribe_client.js)
- [x] by running stress tests

### CODE REVIEW

File: framework/bin/socket_io_subscribe_client.js [hunk 1]
```diff
@@ -68,11 +68,4 @@ const subscribe = subscribeEvent => {
...
 // subscribe('*'); // Listen to all the events
-subscribe('new.block');
-subscribe('new.transactions');
-subscribe('delete.block');
-subscribe('delete.transactions');
-subscribe('update.round');
-subscribe('update.generators');
-subscribe('update.fee_estimates');
-subscribe('update.metadata');
+subscribe('update.index.status');
```
Comment (sameersubudhi) [inline framework/bin/socket_io_subscribe_client.js]: Why are we deleting the other entries?

```suggestion
subscribe('new.block');
subscribe('new.transactions');
subscribe('delete.block');
subscribe('delete.transactions');
subscribe('update.round');
subscribe('update.generators');
subscribe('update.fee_estimates');
subscribe('update.metadata');
subscribe('update.index.status');
```

File: services/gateway/shared/moleculer-io/index.js [hunk 2]
```diff
@@ -443,7 +443,9 @@ function makeHandler(svc, handlerItem) {
 				});
 				svc.logger.info(`   <= ${chalk.green.bold('Success')} ${action}`);
 
-				return addJsonRpcEnvelope(id, res);
+				const output = addJsonRpcEnvelope(id, res);
+				if (respond !== undefined) respond(output);
+				return output;
```
Comment (sameersubudhi) [inline services/gateway/shared/moleculer-io/index.js:448]: Please double check if this change has any side effects, else let's update it to be as below:
```suggestion
				if (respond !== undefined) {
					respond(output);
				} else {
					return output;
				}
				
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/gateway/shared/moleculer-io/index.js (hunk 2 above)
Comment (vardan10) [reply services/gateway/shared/moleculer-io/index.js:448]: Stress tests fail if we don't return output. Hence keeping it as it was.

Comment (sameersubudhi) [reply services/gateway/shared/moleculer-io/index.js:448]: Won't `respond` be `undefined` when we run the stress tests?

Comment (vardan10) [reply services/gateway/shared/moleculer-io/index.js]: My bad, it worked on the second try. Maybe I made some mistake earlier

**Review by sameersubudhi — APPROVED** [review]
Works as expected 👍

File: services/blockchain-indexer/shared/dataService/indexStatus.js [hunk 3]
```diff
@@ -60,9 +47,29 @@ const getIndexStatus = async () => {
...
+Signals.get('indexStatUpdate').add(indexStatUpdateListener);
+Signals.get('numJobsInProgressUpdate').add(indexingProgressListener);
+
 const isBlockchainFullyIndexed = () => Number(indexStatsCache.percentage) === 100;
 
 module.exports = {
 	getIndexStatus,
 	isBlockchainFullyIndexed,
+
+	// Testing
```
Comment (ManuGowda) [inline services/blockchain-indexer/shared/dataService/indexStatus.js:72]: what is this testing?

File: framework/bin/socket_io_subscribe_client.js [hunk 4]
```diff
@@ -68,11 +68,12 @@ const subscribe = subscribeEvent => {
 };
 
 // subscribe('*'); // Listen to all the events
-subscribe('new.block');
```
Comment (ManuGowda) [inline framework/bin/socket_io_subscribe_client.js:71]: is this ok to remove `new.block`?

File: services/blockchain-indexer/shared/dataService/indexStatus.js (hunk 3 above)
Comment (vardan10) [reply services/blockchain-indexer/shared/dataService/indexStatus.js:72]: The below functions are exported for unit tests

File: framework/bin/socket_io_subscribe_client.js (hunk 4 above)
Comment (vardan10) [reply framework/bin/socket_io_subscribe_client.js]: Re-added it

**Review by priojeetpriyom — APPROVED** [review]

**Pull request merged**